src/black/__init__.py

```diff
@@ -1245,8 +1245,7 @@ def get_features_used(  # noqa: C901
             if value_head in {'f"', 'F"', "f'", "F'", "rf", "fr", "RF", "FR"}:
                 features.add(Feature.F_STRINGS)
 
-        elif n.type == token.NUMBER:
-            assert isinstance(n, Leaf)
```

Comment:
So is `TypeGuard` is enforced with mypyc? If not, we're missing an assert. (Just wanting to make sure, since I'm not familiar with this)

Reply:
I'm not familiar with mypyc, so handling over to @ichard26 for this one

Reply:
As I understood [the doc](https://mypy.readthedocs.io/en/latest/type_narrowing.html), `TypeGuard` is not equivalent to `isinstance` at runtime 🤔 Or are we saying that we don't really need it?

Reply:
I don't think it's needed, IIUC the `assert` was only there originally as a hint to `mypy`, it's not a condition one would expect to fail at runtime

Reply:
Oh, I see. If this is the case then consider it resolved!

Reply:
Yeah I don't think it's even possible for this assertion to fail as the CST types representing numbers are always of type `Leaf`. TypeGuard is a cleaner way to tell type checkers that we can conditionally narrow a type.